onnxruntime/core/providers/cuda/nn/pool.cc

```diff
@@ -265,11 +267,17 @@ Status Pool<T, MaxPool<8>, NHWC>::ComputeInternal(OpKernelContext* context) cons
   auto x_data = reinterpret_cast<const CudaT*>(X->Data<T>());
   auto y_data = reinterpret_cast<CudaT*>(Y->MutableData<T>());
 
-  Tensor* I = context->Output(1, TensorShape(y_dims));
+  // I is in NCHW format and the contained indices use NCHW math to compute the index
+  auto i_dims = y_dims;
+  if (NHWC) {
+    std::swap(i_dims[1], i_dims[x_shape.NumDimensions() - 1]);
```

Comment:
This looks like NCHW => NWHC

Reply:
See the comment above for NCHW/NHWC conversion problem